lib/src/ui/puzzle/puzzle_history_screen.dart

```diff
@@ -0,0 +1,263 @@
+import 'package:lichess_mobile/src/widgets/feedback.dart';
+import 'package:lichess_mobile/src/widgets/list.dart';
+import 'package:lichess_mobile/src/widgets/platform.dart';
+import 'package:lichess_mobile/src/widgets/shimmer.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+
+final _puzzleLoadingProvider = StateProvider<bool>((ref) => false);
+
+class PuzzleHistoryWidget extends ConsumerWidget {
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final history = ref.watch(puzzleRecentActivityProvider);
+    return history.when(
+      data: (data) {
+        if (data.isEmpty) {
+          return ListSection(
+            header: Text(context.l10n.puzzleHistory),
+            children: [
+              Center(
+                child: Text(context.l10n.puzzleNoPuzzlesToShow),
+              )
+            ],
+          );
+        }
+
+        return ListSection(
+          header: Text(context.l10n.puzzleHistory),
+          headerTrailing: NoPaddingTextButton(
+            onPressed: () => pushPlatformRoute(
+              context,
+              builder: (context) => PuzzleHistoryScreen(data),
+            ),
+            child: Text(
+              context.l10n.more,
+            ),
+          ),
+          children: [
+            Padding(
+              padding: Styles.bodySectionPadding,
+              child: PuzzleHistoryBoards(data),
+            ),
+          ],
+        );
+      },
+      error: (e, s) {
+        debugPrint(
+          'SEVERE: [PuzzleHistoryWidget] could not load dashboard',
+        );
+        return const Center(child: Text('Could not load Puzzle History'));
+      },
+      loading: () => Shimmer(
+        child: ShimmerLoading(
+          isLoading: true,
+          child: ListSection.loading(
+            itemsNumber: 5,
+            header: true,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class PuzzleHistoryScreen extends StatelessWidget {
+  const PuzzleHistoryScreen(this.historyList);
+
+  final IList<PuzzleHistoryEntry> historyList;
+
+  @override
+  Widget build(BuildContext context) {
+    return PlatformWidget(androidBuilder: _buildAndroid, iosBuilder: _buildIos);
+  }
+
+  Widget _buildIos(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(
+        middle: Text(context.l10n.puzzleHistory),
+      ),
+      child: _Body(historyList),
+    );
+  }
+
+  Widget _buildAndroid(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text(context.l10n.puzzleHistory)),
+      body: _Body(historyList),
+    );
+  }
+}
+
+class _Body extends ConsumerStatefulWidget {
+  const _Body(this.historyList);
+  final IList<PuzzleHistoryEntry> historyList;
+  @override
+  ConsumerState<_Body> createState() => _BodyState();
+}
+
+class _BodyState extends ConsumerState<_Body> {
+  final ScrollController _scrollController = ScrollController();
+  List<PuzzleHistoryEntry> _historyList = [];
```

Comment:
While there is nothing wrong to use a widget state, I think we could improve this by using riverpod to hold the state of the history. (Using a riverpod `Notifier`).
This would allow us to cache the data for some time. Like if a user goes back to the history screen after exiting it, with the widget state we need to fetch all the data again from the server. With riverpod it can be cached.

Also the screen widget should not take the recent items list as a parameter, because that data can be stale.